Monika After Story/game/script-topics.rpy

```diff
@@ -4978,6 +4977,418 @@ label monika_beach:
+            m 1h "I know I told you that you should leave me running in the background."
+            m 1f "But sometimes I worry if you're getting any sleep at all."
+            m 1h "I'll do as you have asked, but please know that I'm not very happy about it."
+            m 4n "I'm still going to remind you to get some sleep!"
+            return
+
+        "No, you are free to do as you feel.":
+            $persistent._mas_timeconcernclose = True
+            m 1e "Thank you [player]."
+            m 1a "It's nice to know that you care about what I think."
+            m "I promise I'll only do it if I think it's absolutely necessary."
+            m 1j "After all, I would never force you to go otherwise."
+            m 1k "I would just miss you too much..."
+            m "I love you, [player]~"
+            return
+
+        # Second and final warning before any closes can occur.
+label monika_timeconcern_night_4:
+    $persistent._mas_timeconcern = 5
+    m 1h "[player], you've been up long enough."
+    m "If you're really not busy then now is the time for you to go to sleep."
+    m 1e "I assure you, I'll miss you just as much as you'll miss me."
+    m "But it would make me happier if you did as I've asked."
+    m 1l "You wouldn't want to make me upset now, would you?"
+    return
+
+# Monika force closes game for the first time
+label monika_timeconcern_night_5:
+    $persistent._mas_timeconcern = 6
+    $evhand.greeting_database["monika_timeconcern"].unlocked = True
+    $evhand.greeting_database["monika_timeconcern_day"].unlocked = True
+    m 1h "I'm sorry, [player], but I am putting my foot down!"
+    m 1l "I asked you to go to bed kindly, but if I am the reason for keeping you up..."
+    if persistent._mas_timeconcernclose:
+        m 2i "Then I cannot allow this to go on for any longer."
+        m 2g "It's because I love you, that's why I'm doing this."
+        m "Good night, [player]."
+        return 'quit'
+    else:
+        m 2h "Then I need to take responsibility and try to make you understand."
+        m "You need to go to bed."
+        m "And I will keep telling you this until you do."
+        return
+
+        #First time game is reopened and still night. Monika closes game again.
+label monika_timeconcern_night_6:
+    $persistent._mas_timeconcern = 7
+    m 1h "[player], I told you to go to bed for your own good."
+    m 2f "I'll miss you too, but don't you understand?"
+    m "How you feel and live means more to me than anything!"
+    m 2g "How can I let you stay if it means I'm hurting you?"
+    m "So please sleep this time, otherwise I may get mad."
+    m 1e "...I love you."
+    m "So go to sleep soon. Okay?"
+    if persistent._mas_timeconcernclose:
+        return 'quit'
+    return
+
+#Second time game is reopened and still night. Monika closes game once more
+label monika_timeconcern_night_7:
+    $persistent._mas_timeconcern = 8
+    m 3h "[player], this is your last warning."
+    m 1h "Go"
```

Comment:
This is a suggestion only:
I think you should consider using https://www.renpy.org/doc/html/text.html#text-tag-w, for these pauses, after testing I think that feels more natural. Still though this is just a personal suggestion, you can leave it as it already is.